src/middleware.ts

```diff
@@ -8,6 +8,11 @@ export const config = {
 export default withAuth(async (req) => {
   if (process.env.LOCAL_CMS_PROVIDER) return;
   const token = req.nextauth.token;
+  const userIp =
+    req.headers.get('x-forwarded-for') ||
+    req.ip ||
+    req.headers.get('x-real-ip');
+
   if (!token) {
     return NextResponse.redirect(new URL('/invalidsession', req.url));
   }
```

Comment:
In this endpoint itself, send the ip and do the check there
Dont do it saparately below


Or


move all the logic for token check here as well

Dont have two saparate places for doing the check

Reply:
@hkirat done